lecture-classification: incorporated feedback and updated to have a more rounded lecture

melsigl opened the pull request:
Incorporated feedback that has been provided in commit https://github.com/FAU-CS6/KDD/commit/d633a03e0e92eb72a9ddcc0186f034e25409598f. Additionally, updated lecture to include more detailed algorithm descriptions of:
- decision tree (detailed algorithm is in appendix)
- updated algorithm for AdaBoost

Closes issue #6.

dominik-probst reviewed (approved):
Approved as this PR is mainly the implementation of feedback from a previous review.

melsigl commented:
I know that giving feedback/review to this lecture has been far from the standard procedure as the lecture has already been merged into our `dev` branch. However, dealing with every single review in code has been a nuisance for the following reasons, which I hope you'll understand and you do not get me wrong:
- Merely informational review comments must be deleted. That's an extra effort that could be saved.
- All transparency and discussions as to why I incorporated some feedback and how I incorporated that feedback is lost. I managed to at least retain some transparency by creating a single commit (or cluster smaller changes) into one commit and moved my explanation to the commit message. I am afraid that is not the best solution here, but given the nature of the review hopefully acceptable and understandable.
- Changes to definition blocks have been made. Some of them wrongly. For instance in a headline it is common to capitalize words such as "Decision Tree", "Naive Bayes", or "Gini Index". In the text, however, it is common to not capitalize these words, except these are introduced the first time and a abbreviation exists, or it contains a persons name (like it is the case with naive Bayes and Gini index). Removing these changes also took time which could've been avoided if there has been an overal information that these should be changed.

All in all, using Githubs' review process retains transparency and discussion, which, again, i know has not been possible in this concrete example (as mentioned above). 

I truly appreciate your time and effort in reviewing this lecture. Do not let my little rant here tell you otherwise.
